src/resolver.rs

```diff
@@ -341,6 +341,40 @@ pub enum RequiredEntry {
     },
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+enum DeltaEdgeFreshness {
+    // All information requried for this delta edge is already stored within the store's
+    // supply-chain. Using this edge will require no changes to imports.lock.
+    Stale,
+    // This edge originates from an already-imported or local wildcard audit, however will require
+    // importing fresh publisher information into imports.lock.
+    FreshPublisher,
+    // This edge is fully fresh, and will require adding the audit entry to imports.lock to use.
+    Fresh,
+}
+
+impl DeltaEdgeFreshness {
+    pub fn fresh(is_fresh: bool) -> Self {
+        if is_fresh {
+            DeltaEdgeFreshness::Fresh
+        } else {
+            DeltaEdgeFreshness::Stale
+        }
+    }
+
+    pub fn fresh_publisher(is_fresh_publisher: bool) -> Self {
+        if is_fresh_publisher {
+            DeltaEdgeFreshness::FreshPublisher
+        } else {
+            DeltaEdgeFreshness::Stale
+        }
+    }
```

Comment:
Perhaps this one should handle both sides, rather than just the publisher one?

Also not sure why these methods are public given the type they are on is not exposed outside of this module, so let's drop the `pub` annotations.

```suggestion
   fn fresh_publisher(is_fresh_audit: bool, is_fresh_publisher: bool) -> Self {
        if is_fresh_audit {
            DeltaEdgeFreshness::Fresh
        } else if is_fresh_publisher {
            DeltaEdgeFreshness::FreshPublisher
        } else {
            DeltaEdgeFreshness::Stale
        }
    }
```